src/state/client-server-hooks/use-station-filtering-defaults.js

```diff
@@ -0,0 +1,35 @@
+import { useEffect } from "react";
+import { useShallow } from "zustand/react/shallow";
+import { useVariables } from "@/state/query-hooks/use-variables";
+import { useFrequencies } from "@/state/query-hooks/use-frequencies";
+import { useNetworks } from "@/state/query-hooks/use-networks";
+import { useStationsStore } from "@/state/client/stations-store";
+import map from "lodash/fp/map";
+import identity from "lodash/fp/identity";
+
+/**
+ * Integration between query hooks loaded on the Body route and setting of defaults for filters.
+ * @returns
+ */
+export const useStationFilteringDefaults = () => {
+  const { data: variables } = useVariables();
+  const { data: frequencies } = useFrequencies();
+  const { data: networks } = useNetworks();
+  const { setDefaultNetworks, setDefaultVariables, setDefaultFrequencies } =
+    useStationsStore(
+      useShallow((state) => ({
+        setDefaultNetworks: state.setDefaultNetworks,
+        setDefaultVariables: state.setDefaultVariables,
+        setDefaultFrequencies: state.setDefaultFrequencies,
+      })),
+    );
+
+  useEffect(() => {
```

Comment:
Is this just being used for the effect of memoization? Would `useMemo` be preferable?

Reply:
That is correct, but as the function doesn't return anything (we're not storing anything, it sets the other state properties) I think useEffect is more appropriate in this case.